### PR TITLE
Adding Phase2 tracker T36, T37 and T38 (for IT studies)

### PR DESCRIPTION
**Pull request by Raffaella07**
#### PR description:

This PR introduces three new phase-2 Tracker layouts as presented in:
https://indico.cern.ch/event/1344694/contributions/5725063/attachments/2775549/4836942/DPG_TrackerSim_updates.pdf

- T36 merges previous developments on TFPX yawned geometry (T30, introduced in #36660) and TBPX split sensors (T33, introduced in #41880, fixed as T33/T35 in #43712 ) 
- T37 inherits from T36 + the Z+ and Z- rods of TBPX layers are shifted of 0.7 mm each towards the endcaps.
- T38  inherits from T36 + the Z+ and Z- rods of TBPX layers are shifted of 1.3 mm each towards the endcaps.

T36:
Geometry scenario: 2026D111.
Workflows: 300xx .0 

T37:
Geometry scenario: 2026D112.
Workflows: 304xx.0

T38:
Geometry scenario: 2026D113.
Workflows: 308xx.0

#### PR validation:

- Checked XMLs make sense from Tracker design point of view
- Checked workflow numbering
- Checked that workflows with scenarios D111, D112 and D113 run smoothly with no errors
- Warnings on L1NNCaloTauEmulator - BrokenTowers (step2) DQMGenericClient:postProcessorHLTmuonTracking@endRun  (step4). Should not depend on tracker geometry, same exact warnings are present in D108 (tested wf 28824) in clean CMSSW_14_1_pre2. 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

not a backport 

@emiglior @mmusich @bsunanda @sroychow @ FYI

### CODE REVIEW

**Comment by mmusich** [comment]
test parameters:
   * workflows = 22034.0, 30034 .0 30434.0, 30034.0

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -1654,13 +1660,12 @@ def setup_(self, step, stepName, stepDict, k, properties):
 
 class UpgradeWorkflow_ProdLike(UpgradeWorkflow):
     def setup_(self, step, stepName, stepDict, k, properties):
-        thisStep = stepDict[step][k]["-s"]
         if 'GenSimHLBeamSpot14' in step:
             stepDict[stepName][k] = merge([{'--eventcontent': 'RAWSIM', '--datatier': 'GEN-SIM'},stepDict[step][k]])
         elif 'Digi' in step and 'Trigger' not in step:
-            stepDict[stepName][k] = merge([{'-s': thisStep.replace("DIGI:pdigi_valid","DIGI"),'--datatier':'GEN-SIM-RAW', '--eventcontent':'RAWSIM'}, stepDict[step][k]])
+            stepDict[stepName][k] = merge([{'-s': 'DIGI,L1,DIGI2RAW,HLT:@relval2022', '--datatier':'GEN-SIM-RAW', '--eventcontent':'RAWSIM'}, stepDict[step][k]])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:1666]: mh, this seems to have reverted the changes that entered in https://github.com/cms-sw/cmssw/pull/44525.
@Raffaella07 please revert these changes.

**Comment by mmusich** [comment]
@cmsbuild, please abort

File: Geometry/TrackerNumberingBuilder/README.md [hunk 2]
```diff
@@ -195,6 +195,31 @@ Only the subdetector 2 schema is different wrt the original Phase 2 upgrade DetI
 The configuration names for this detid schema are `trackerNumberingGeometry_cfi` (to run on geometry built from xml files) or `trackerNumberingGeometryDB_cfi` (to run on geometry from DB) for `TrackerGeometricDetESModule` and `trackerTopology2023Constants_cfi` for `TrackerTopology`
 The xml description of tracker parameters for this detid schema is in [Geometry/TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml](../TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml
 
+### Phase 2 Upgrade Detector DetId schema modified to account for the re-ordering of the pixel + L1 splitted sensor
```
Comment (mmusich) [inline Geometry/TrackerNumberingBuilder/README.md:198]:
```suggestion
### Phase 2 Upgrade Detector DetId schema modified to account for the re-ordering of the pixel + L1 split sensor 
```

File: Geometry/TrackerNumberingBuilder/README.md [hunk 3]
```diff
@@ -195,6 +195,31 @@ Only the subdetector 2 schema is different wrt the original Phase 2 upgrade DetI
 The configuration names for this detid schema are `trackerNumberingGeometry_cfi` (to run on geometry built from xml files) or `trackerNumberingGeometryDB_cfi` (to run on geometry from DB) for `TrackerGeometricDetESModule` and `trackerTopology2023Constants_cfi` for `TrackerTopology`
 The xml description of tracker parameters for this detid schema is in [Geometry/TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml](../TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml
 
+### Phase 2 Upgrade Detector DetId schema modified to account for the re-ordering of the pixel + L1 splitted sensor 
+
+From CMSSW_14_1_X on [Geometry/TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml](../TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml has been updated to reflect a single geometry merging the splitted sensors detID numbering and the detID rearrangement described above.
```
Comment (mmusich) [inline Geometry/TrackerNumberingBuilder/README.md:200]:
```suggestion
From CMSSW_14_1_X on [Geometry/TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml](../TrackerCommonData/data/PhaseII/TFPXTEPXReordered/trackerParameters.xml) has been updated to reflect a single geometry merging the splitted sensors detID numbering and the detID rearrangement described above. 
```

Comment (emiglior) [reply Geometry/TrackerNumberingBuilder/README.md:200]: @gbardell @langufo FYI

**Comment by emiglior** [comment]
about the potential conflicts with [PR 44576](https://github.com/cms-sw/cmssw/pull/44576), my preference would be to give higher priority to this PR

File: Configuration/Geometry/README.md [hunk 4]
```diff
@@ -161,3 +161,6 @@ Several detector combinations have been generated:
 * D108 = T35+C19+M11+I17+O9+F8
 * D109 = T35+C24+M11+I17+O9+F8
 * D110 = T35+C18+M11+I17+O9+F8
+* D111 = T36+C24+M11+I17+O9+F8
+* D110 = T37+C24+M11+I17+O9+F8
+* D110 = T38+C24+M11+I17+O9+F8
```
Comment (perrotta) [inline Configuration/Geometry/README.md:166]: There are two "D110" too much
```suggestion
* D112 = T37+C24+M11+I17+O9+F8
* D113 = T38+C24+M11+I17+O9+F8
```

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 5]
```diff
@@ -3057,6 +3063,28 @@ def condition(self, fragment, stepList, key, hasHarvest):
...
+        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal', 'ALCAPhase2'],
+    },
+    '2026D113' : {
+        'Geom' : 'Extended2026D113',
+        'HLTmenu': '@relval2026',
+        'GT' : 'auto:phase2_realistic_T38',
+        'Era' : 'Phase2C22I13M9',
+        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal', 'ALCAPhase2'],
+    },
+}
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:3087]: Extra `}` ?

Comment (Raffaella07) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:3087]: Removed } on line 3088, was mistakenly inserted in an incorrect squash attempt, thanks for the flash review!

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by srimanob** [comment]
The workflow 30834.0 is missing from the test due to typo. I will retrigger the test again.

**Comment by srimanob** [comment]
test parameters:

- workflow_opts = -w upgrade
- workflows = 22034.0, 30034 .0, 30434.0, 30834.0

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by emiglior** [comment]
Hi, as providing these geometries is one of the items to be reported at the forthcoming TK Annual Review (Apr 29th) it would be really nice to have the PR fully signed by then...

**Comment by mmusich** [comment]
@cms-sw/orp-l2 your signature is required here.
